components/esp-tls/esp_tls_mbedtls.c

```diff
@@ -566,7 +566,16 @@ esp_err_t set_server_config(esp_tls_cfg_server_t *cfg, esp_tls_t *tls)
             return esp_ret;
         }
     } else {
+#if defined(CONFIG_ESP_TLS_SERVER_SNI_HOOK)
+        if (cfg->sni_callback == NULL) {
```

Comment:
I think the SNI callback needs to be configured here.
I think only the check if sni_callback is present needs to be in the `else` part.
We should allow users to set the sni_callback along with providing server cert and keys. I think the sni callback would only be triggered when the ClientHello message would contain the `ServerName` extension. Otherwise, the normal certs would be used.